UserDB.cpp

```diff
@@ -65,6 +74,20 @@ void UserDB::Start()
         std::thread update_thread([this](){ this->update_task(); });
         _inotify_thread = std::move(inotify_thread);
         _update_thread = std::move(update_thread);
+
+        int ret = sd_bus_open_system(&bus);
+        if (ret < 0) {
+            Logger::Error("Failed to connect to system bus: %s", strerror(-ret));
+        }
+
+        // Initialize the user list
+        update_user_list();
+
+        // Start the listener thread for user change signals
+        listener_thread = std::thread(&UserDB::ListenForUserChanges, this);
+
+        // Wait for a short period to allow the listener to initialize and capture signals
```

Comment:
>Wait for a short period to allow the listener to initialize and capture signals [](http://example.com/codeflow?start=11&length=79)

What if the initialization takes more time?

Reply:
Added a poller implementation to check every 5 secs until 30 secs.